js/modules/WebAPIProvider/AuthAPI.js

```diff
@@ -25,6 +32,33 @@ define(function(require, exports) {
         return new Date(expirationInSeconds * 1000);
 
     });
+
+    const tokenExpired = ko.observable(false);
+    const askLoginOnTokenExpire = (function() {
+        let expirationTimeout;
+        return () => {
+            if (expirationTimeout) {
+                clearTimeout(expirationTimeout);
+            }
+            if (tokenExpirationDate() > new Date()) {
+                tokenExpired(false);
+                expirationTimeout = setTimeout(
+                    () => {
+                        tokenExpired(true);
+                        $('#myModal').modal('show');
```

Comment:
The login modal is already bound to `<div data-bind="if: !isLoggedIn() || tokenExpired()">`, why do we need to show this modal?

Reply:
@johnSamilin , where is the modal bound to the condition? Its contrnt does